twitterscraper/TwitterScraper.py

```diff
@@ -2,147 +2,166 @@
+        if (self.collected_tweets < self.no_tweets) and len(tweets) > 0:
+            return True
+        else:
+            return False
 
-	def parse_recipients(*recipients):
-		if recipients[1]:
-			if type(recipients[1]) is str:
-				recipients_string = recipients[1]
-			elif type(recipients[1]) is list:
-				recipients_string = "%20OR%20to%3A".join(recipients[1])
-			return recipients_string
+    def parse_url(self):
+        url_1 = "https://twitter.com/search?f=tweets&vertical=default&q="
+        url_2 = "https://twitter.com/i/search/timeline?f=tweets&vertical=default&include_available_features=1&include_entities=1&reset_error_state=false&src=typd"
+        if self.is_first_iteration():
+            url = url_1 + self.topics
+        else:
+            url = url_2 + \
+                "&max_position=%s&q=%s" % (self.min_position, self.topics)
+        if self.lang:
+            url += "%20lang%3A" + self.lang
+        if self.begin_date:
+            url += "%20since%3A" + self.begin_date
+        if self.end_date:
+            url += "%20until%3A" + self.end_date
+        if self.authors:
+            url += "%20from%3A" + self.authors
+        if self.recipients:
+            url += "%20to%3A" + self.recipients
+        if self.location:
+            url += '%20' + self.location
+        return url
 
-	def parse_location(self, location, within):
-		if location:
-			if type(location) is str:
-				location_string = 'near%3A"' + location + '"%20within%3A' + within
-			elif type(location) is list:
-				location_string = '"geocode%3A' + str(location[0]) + '%2C' + str(location[1]) + '%2C' + within
-			return location_string
-		
-	def is_first_iteration(self):
-		return True if self.min_position == -1 else False
+    def scrape_tweets(self):
+        url = self.parse_url()
+        headers = {'User-Agent': random.choice(headers_list)}
+        req = urllib.request.Request(url, headers=headers)
+        tweets = []
+        try:
+            response = urllib.request.urlopen(req).read().decode()
+            if self.is_first_iteration():
+                html = response
+            else:
+                response_json = json.loads(response)
+                html = response_json['items_html']
+            soup = BeautifulSoup(html, "lxml")
+            tweets = soup.find_all('li', 'js-stream-item')
+            if tweets:
+                self.last_tweet_id = tweets[-1]['data-item-id']
+                self.first_tweet_id = tweets[0]['data-item-id']
+                if self.is_first_iteration():
+                    self.min_position = "TWEET-%s-%s" % (self.last_tweet_id,
+                                                         self.first_tweet_id)
+                else:
+                    minp_splitted = response_json['min_position'].split('-')
+                    minp_splitted[1] = self.last_tweet_id
+                    self.min_position = "-".join(minp_splitted)
+        except urllib.request.HTTPError as e:
```

Comment:
If only urlopen and Request are imported these exceptions also need to be imported separately, or removed.  (latter seems better to me).